### ISSUE TITLE
Thread Safety

### ISSUE THREAD

**Issue by Downsite** [ERGO-Code/HiGHS#810]
I am on commit  "71451792f302a4d38dfa90d1071491116306d797" (the last time I tried to update, my CMake script was unable to find the HiGHS headers suddenly).

When I try to run different instances of HiGHS inside of std::async, I receive a SEGVAULT inside the file

```
Thread 3 "X" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffeffff640 (LWP 48347)]
0x0000555555c12b6a in HighsSplitDeque::getCurrentHead (this=0x0) at /home/X/dep/HiGHS/src/./parallel/HighsSplitDeque.h:481
481	  int getCurrentHead() const { return ownerData.head; }
(gdb) bt
#0  0x0000555555c12b6a in HighsSplitDeque::getCurrentHead (this=0x0) at /home/X/dep/HiGHS/src/./parallel/HighsSplitDeque.h:481
#1  0x0000555555c12d99 in highs::parallel::TaskGroup::TaskGroup (this=0x7fffefff85e0) at /home/X/dep/HiGHS/src/./parallel/HighsParallel.h:75
#2  0x0000555555c0be61 in HighsMipSolverData::evaluateRootNode (this=0x7fffe809b140) at /home/X/dep/HiGHS/src/mip/HighsMipSolverData.cpp:1198
#3  0x0000555555bf0ae7 in HighsMipSolver::run (this=0x7fffefff90d0) at /home/X/dep/HiGHS/src/mip/HighsMipSolver.cpp:122
#4  0x0000555555b8f499 in Highs::callSolveMip (this=0x7fffefff9d90) at /home/X/dep/HiGHS/src/lp_data/Highs.cpp:2761
#5  0x0000555555b84afa in Highs::run (this=0x7fffefff9d90) at /home/X/dep/HiGHS/src/lp_data/Highs.cpp:804
```

Interestingly, I also encounter the same problem if I encapsulate all calls to HiGHS functions in a single function and use std::lock_guard to avoid it being called by multiple threads at the same time.

Outside of the std::async, I got the following error trying to call the HiGHS-containting-function (still with barriers around) from different threads with different settings for threads.

`Option 'threads' is set to 2 but global scheduler has already been initialized to use 1 threads. The previous scheduler instance can be destroyed by calling HighsTaskExecutor::shutdown().`

Is there something about using HiGHS in parallel that I should be aware of?

**Comment by Downsite** [comment]
After realizing, that I needed to set the FAST_BUILD flag to make CMake work again, I tested on the current master, with the same results.

**Comment by lgottwald** [comment]
indeed I overlooked something for the use of multiple Highs instances with multi-threading. Will think about how to address this.

**Comment by lgottwald** [comment]
Could you test if the error is resolved in the branch of fix-810 in the pull request #818

**Comment by Downsite** [comment]
Not sure how complete the test case is, but before, it would crash every time and on the fix-810 branch, there are no longer crashes occuring.
